File: src/main/java/org/homework20241215/ramen/Ramen.java
package org.homework20241215.ramen;

import java.util.HashMap;
import java.util.Map;

public class Ramen {

    public static RamenBuilder builder() {
        return new RamenBuilder();
    }

    private final String name;
    private final Integer ramenPrice;
    private Map<Topping, Integer> toppings = new HashMap<>();

    private int totalPrice = 0;

    public Ramen(String name, int ramenPrice, Map<Topping, Integer> toppings) {
        this.name = name;
        this.ramenPrice = ramenPrice;
        this.toppings = toppings;

        totalPrice = ramenPrice;
        toppings.forEach((topping, price) -> totalPrice += price);

    }

    public void printRamenDetail() {
        System.out.println(name + "(" + totalPrice + "円)");
        System.out.println("トッピング");
        toppings.forEach((topping, price) -> System.out.println(topping));
    }


}

File: src/main/java/org/homework20241222/ramen/Ramen.kt
package org.homework20241222.ramen

class Ramen(private val name: String, val ramenPrice: Int, private val toppings: Map<Topping, Int>){
    companion object {
        fun builder(): RamenBuilder {
            return RamenBuilder()
        }
    }

    private var totalPrice: Int = 0;

    fun printRamenDetail(ramenPrice: Int) {
        totalPrice = ramenPrice
        toppings.forEach{(topping, price) -> totalPrice += price}
        println(name + "(" + totalPrice + "円)")
        println("トッピング")
        toppings.forEach{(topping, price) -> println(topping)}
    }


}
